domenetjenester/brevbestiller/src/main/java/no/nav/foreldrepenger/melding/brevmapper/brev/opphørSVP/OpphørsperiodeMapper.java

```diff
@@ -0,0 +1,119 @@
+package no.nav.foreldrepenger.melding.brevmapper.brev.opphørSVP;
+
+import no.nav.foreldrepenger.melding.Tuple;
+import no.nav.foreldrepenger.melding.behandling.Behandling;
+import no.nav.foreldrepenger.melding.behandling.Behandlingsresultat;
+import no.nav.foreldrepenger.melding.datamapper.domene.BehandlingMapper;
+import no.nav.foreldrepenger.melding.datamapper.domene.FellesMapper;
+import no.nav.foreldrepenger.melding.datamapper.domene.sortering.LovhjemmelComparator;
+import no.nav.foreldrepenger.melding.fagsak.FagsakYtelseType;
+import no.nav.foreldrepenger.melding.geografisk.Språkkode;
+import no.nav.foreldrepenger.melding.integrasjon.dokgen.dto.felles.Årsak;
+import no.nav.foreldrepenger.melding.integrasjon.dokgen.dto.opphørSVP.OpphørPeriode;
+import no.nav.foreldrepenger.melding.uttak.PeriodeResultatType;
+import no.nav.foreldrepenger.melding.uttak.svp.ArbeidsforholdIkkeOppfyltÅrsak;
+import no.nav.foreldrepenger.melding.uttak.svp.PeriodeIkkeOppfyltÅrsak;
+import no.nav.foreldrepenger.melding.uttak.svp.SvpUttakResultatArbeidsforhold;
+import no.nav.foreldrepenger.melding.uttak.svp.SvpUttakResultatPeriode;
+import no.nav.foreldrepenger.melding.vilkår.Avslagsårsak;
+import no.nav.foreldrepenger.melding.vilkår.VilkårType;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public class OpphørsperiodeMapper {
+    private static Set<String> lovReferanser;
+    
+    public static Tuple<List<OpphørPeriode>, String> mapOpphørtePerioderOgLovhjemmel(Behandling behandling, FagsakYtelseType ytelseType, List<SvpUttakResultatArbeidsforhold> uttakResultatArbeidsforhold, Språkkode språkKode) {
+        lovReferanser = new TreeSet<>(new LovhjemmelComparator());
+        Behandlingsresultat behandlingsresultat = behandling.getBehandlingsresultat();
+        Avslagsårsak avslagsårsak = behandlingsresultat.getAvslagsårsak();
+
+        //Henter opphørte perioder fra uttak - hvis finnes.
+        List<OpphørPeriode> opphørtePerioder = mapOpphørteUttaksperioder(uttakResultatArbeidsforhold, språkKode);
+
+        //I noen tilfeller er behandlingen opphørt før det har vært uttak
+        if(opphørtePerioder.isEmpty()) {
+              if(avslagsårsak != null) {
+                  opphørtePerioder = mapOpphørteUttaksperioderFraInnvilget(uttakResultatArbeidsforhold, språkKode, avslagsårsak.getKode());
+                  if (opphørtePerioder.isEmpty()) {
+                      opphørtePerioder.add(opprettSvpOpphørUtenPeriode(avslagsårsak.getKode(), Collections.emptyList()));
+                  }
+                  lovReferanser.add(leggTilLovreferanse(avslagsårsak, ytelseType));
+              } else {
+                  //Det er også mulig at opphørsårsak finnes på arbeidsforholdet?? To do Anja: Avklare om det faktisk skal brukes
```

Comment:
TODO bør alltid skrives som `//TODO(Anja)`, så vil de plukkes opp av IntelliJ sin TODO-funksjon, og/eller man kan søke på dem enkelt for å rydde de som er glemte i ettertid.

Reply:
Utført